Pritipathak/update api doc (#64)
microsoft/carbon-aware-sdk: src/dotnet/CarbonAware.WebApi/README.md

> The Carbon Aware SDK provides api to get the carbon intensity for a location at a given time period. In order to use the following API, the environment has to be setup with the right configuration. Instructions on setting up the environment could be found here - https://github.com/microsoft/carbon-aware-sdk/blob/dev/GettingStarted.md

Reviewer: ```suggestion
The Carbon Aware SDK provides an API to get the Green Software Foundation's definition for carbon intensity given a location and time period. In order to use the API, the environment needs to be prepared with a set of configuration parameters. Instructions on setting up the environment could be found here - https://github.com/microsoft/carbon-aware-sdk/blob/dev/GettingStarted.md
```

> # Carbon Aware REST API

Reviewer: This is great verbiage here that really helps drive clarity on what each API does - would it make sense to move this into the decorators on each of the calls in the controller code so that the API spec gets populated with this information?

Author: IMO, it would be redundant to have this information on each call. I think each API call should ideally only document what the call does with information about the input, output payload.

Reviewer 2: I agree with George.  Swagger will list this information and make it easy to submit sample calls.

We should also tell them how to get to the swagger endpoint, once they have the WebAPI running.

Author: Sounds good.

> The Carbon Aware SDK provides an API to get the Green Software Foundation's definition for carbon intensity given a location and time period. In order to use the API, the environment needs to be prepared with a set of configuration parameters. Instructions on setting up the environment could be found here - https://github.com/microsoft/carbon-aware-sdk/blob/dev/GettingStarted.md

Reviewer 3: ```suggestion
The Carbon Aware SDK provides an API to get the marginal carbon intensity for a given location and time period. The values reported in the Green Software Foundation's specification for marginal carbon intensity (Grams per Kilowatt Hour). In order to use the API, the environment needs to be prepared with a set of configuration parameters. Instructions on setting up the environment could be found here - https://github.com/microsoft/carbon-aware-sdk/blob/dev/GettingStarted.md
```

}
```

### GET emissions/bylocation

> This endpoint calculates the observed emission data by location for a specified time period

Reviewer 3: ```suggestion
This endpoint calculates the observed emission data by location for a specified time period.
```

> location is a required parameter and is name of a data region for the given Cloud provider.

Reviewer 3: ```suggestion
Location is a required parameter and is name of the data region for the configured Cloud provider.
```

> If time period is not provided, it retrieves all the data until the current time.

Reviewer 3: There is some nuance here.  `All available data` might be more accurate, though for WattTime that's still not technically accurate because it wont include historical data :)

EG
```
https://<server_name>/emissions/bylocation?location=useast&time=2022-01-01&toTime=2022-05-17
```

> The response is EmissionsData object that contains the location, time and the rating in Gms/KWH

Reviewer 3: ```suggestion
The response is an array of EmissionsData objects that contains the location, time and the rating in g/kWh
```

EG
```

> {

Reviewer 3: ```suggestion
[
  {
```

  "location":"eastus",
  "time":"2022-05-17T20:45:11.5092741+00:00",
  "rating":70

> }

Reviewer 3: ```suggestion
  }
]
```

```

### GET emissions/bylocations

This endpoint calculates the observed emission data by list of locations for a specified time period

> location is a required parameter and is a list of data regions for the given Cloud provider.

Reviewer 3: Same tweak as above

If time period is not provided, it retrieves all the data until the current time.

EG
```

> https://<server_name>/emissions/bylocation?locations=eastus&locations=westus&time=2022-01-01&toTime=2022-05-17

Reviewer 3: ```suggestion
https://<server_name>/emissions/bylocations?locations=eastus&locations=westus&time=2022-01-01&toTime=2022-05-17
```

```

> The response is list of EmissionsData objects that contains the location, time and the rating in Gms/KWH

Reviewer 3: ```suggestion
The response is an array of EmissionsData objects that contains the location, time and the rating in g/kWh.
```

}
```

### GET emissions/bylocation

This endpoint calculates the observed emission data by location for a specified time period

> The payload object must include location. 

Reviewer 4: Correction: This route (GET emissions/bylocation) doesn't receive a payload.

If location is of type `CloudProvider`, the object should include the `providerName` and `regionName` attributes.
If location if of type `Geoposition` then the object should include `latitude` and `longitude` attributes.
If time period is not provided, it retrieves all the data until the current time.

EG
```
{
    "location": {
        "locationType": "CloudProvider",
        "providerName": "Azure",
        "regionName": "useast"
    },
    "startTime": "2020-03-01T13:00:00Z",
    "endTime": "2021-03-01T15:30:00Z"
}
```

The response is EmissionsData object that contains the location, time and the rating in Gms/KWH
EG
```
{
  "location":"eastus",
  "time":"2022-05-17T20:45:11.5092741+00:00",
  "rating":70
}

```

> ### GET emissions/bylocations

Reviewer 4: Same comment as the other GET

```

The response is an array of EmissionsData objects that contains the location, time and the rating in g/kWh.
EG
```
[
  {
    "location":"eastus"
    "time":"2022-05-17T20:45:11.5092741+00:00",
    "rating":70
  }
]
```

### GET emissions/bylocations/best

This endpoint calculates the best observed emission data by list of locations for a specified time period

Location is a required parameter and is name of the data region for the configured Cloud provider.
If time period is not provided, it retrieves all the data until the current time.

EG
```
https://<server_name>/emissions/bylocations/best?locations=eastus&locations=westus&time=2022-01-01&toTime=2022-05-17
```

Currently, the response is an array of EmissionsData objects that contains the location, time and the rating in g/kWh.

> This response will be updated in the furture to contain a single EmissionsData object with location, time and the rating in g/kWh. 

Reviewer 3: NIT: I would remove and only document current functionality.  Without a timeline for when this proposed change will be implemented, I think this might breed more confusion than clarity.
